OpenRA.Mods.Common/Traits/Render/RenderSprites.cs

```diff
@@ -236,19 +237,38 @@ public static string NormalizeSequence(Animation anim, DamageState state, string
 			return sequence;
 		}
 
-		// Required by WithSpriteBody and WithInfantryBody
 		public int2 AutoSelectionSize(Actor self)
 		{
-			return AutoRenderSize(self);
+			var visibleAnims = anims.Where(b => b.IsVisible && b.Animation.Animation.CurrentSequence != null).ToArray();
+			var aggregateSize = int2.Zero;
+			foreach (var a in visibleAnims)
+			{
+				var size = (a.Animation.Animation.Image.Size.XY * info.Scale).ToInt2();
+				if (size.X * size.X > aggregateSize.X * aggregateSize.X)
+					aggregateSize = new int2(size.X, aggregateSize.Y);
+				if (size.Y * size.Y > aggregateSize.Y * aggregateSize.Y)
+					aggregateSize = new int2(aggregateSize.X, size.Y);
+			}
+
+			return aggregateSize;
 		}
 
-		// Required by WithSpriteBody and WithInfantryBody
-		public int2 AutoRenderSize(Actor self)
+		Rectangle IRender.AutoRenderBounds(Actor self)
```

Comment:
This could then become something like
```csharp
IEnumerable<Rectangle> ScreenBounds(Actor self, WorldRenderer wr)
{
	foreach (var a in anims)
	{
		if (!a.IsVisible)
			continue;

		foreach (var r in a.Animation.Render(self, wr, null, info.Scale))
		{
			if (!(r is SpriteRenderable))
				continue;

			// NOTE: This could be simplified if we define ScreenBounds on IRenderable
			yield return ((SpriteRenderable)r).ScreenBounds(wr);
		}
	}
}
```